GraphBulkImporter/Program.cs

```diff
@@ -0,0 +1,194 @@
+
+        private static readonly ConnectionPolicy ConnectionPolicy = new ConnectionPolicy
+        {
+            ConnectionMode = ConnectionMode.Direct,
+            ConnectionProtocol = Protocol.Tcp
+        };
+
+        private static DocumentClient client;
+
+        public async static Task Main(string[] args)
+        {
+            Trace.WriteLine("Summary:");
+            Trace.WriteLine("--------------------------------------------------------------------- ");
+            Trace.WriteLine(String.Format("Endpoint: {0}", EndpointUrl));
+            Trace.WriteLine(String.Format("Collection : {0}.{1}", DatabaseName, CollectionName));
+            Trace.WriteLine("--------------------------------------------------------------------- ");
+            Trace.WriteLine("");
+
+            try
+            {
+                using (client = new DocumentClient(
+                    new Uri(EndpointUrl),
+                    AuthorizationKey,
+                    ConnectionPolicy))
+                {
+                    await RunBulkImportAsync();
+                }
+            }
+            catch (AggregateException e)
+            {
+                Trace.TraceError("Caught AggregateException in Main, Inner Exception:\n" + e);
+                Console.ReadKey();
+            }
+
+        }
+
+        private async static Task RunBulkImportAsync()
+        {
+            // Cleanup on start if set in config.
+            DocumentCollection dataCollection = null;
+            try
+            {
+                if (bool.Parse(ConfigurationManager.AppSettings["ShouldCleanupOnStart"]))
+                {
+                    Database database = Utils.GetDatabaseIfExists(client, DatabaseName);
+                    if (database != null)
+                    {
+                        await client.DeleteDatabaseAsync(database.SelfLink);
+                    }
+
+                    Trace.TraceInformation("Creating database {0}", DatabaseName);
+                    database = await client.CreateDatabaseAsync(new Database { Id = DatabaseName });
+
+                    Trace.TraceInformation(String.Format("Creating collection {0} with {1} RU/s", CollectionName, CollectionThroughput));
+                    dataCollection = await Utils.CreatePartitionedCollectionAsync(client, DatabaseName, CollectionName, CollectionThroughput);
+                }
+                else
+                {
+                    dataCollection = Utils.GetCollectionIfExists(client, DatabaseName, CollectionName);
+                    if (dataCollection == null)
+                    {
+                        throw new Exception("The data collection does not exist");
+                    }
+                }
+            }
+            catch (Exception de)
+            {
+                Trace.TraceError("Unable to initialize, exception message: {0}", de.Message);
```

Comment:
Message might be fine in this case if it is being logged elsewhere, but I always find de.ToString() to be more useful because it actually contains the stacktrace